libvast/src/system/pivoter.cpp

```diff
@@ -103,17 +103,17 @@ caf::behavior pivoter(caf::stateful_actor<pivoter_state>* self, caf::actor node,
         return;
       VAST_DEBUG(self, "uses", *pivot_field, "to extract", st.target, "events");
       auto column = slice->column(pivot_field->name);
-      auto xs = set{};
+      auto xs = vector{};
```

Comment:
This looks like a bug to me.

Reply:
Only if the input to `xs` is not unique.

Reply:
```suggestion
      auto xs = detail::stable_set<data>{};
```

And then switch back to insert below.

Reply:
Does it need to be an order-preserving set or would an unordered set work as well?

Reply:
I'd argue that for bisecting reasons (if we were to find a bug with this), you should just use what I proposed, as it essentially is `vast::set`. This PR removes the set data type, it should not change the way pivoting works.

Reply:
The thing is, I need to create a `data` from the `vector` later on. So if I use a stable set, I need to move all items over into a new container.

I looked at the code again and there is no difference to using a `vector` here, because uniqueness is enforced via the auxiliary structure `requested_ids`. So using `vector` instead of `set` is safe.

Reply:
If you are certain that this does not change the behavior, feel free to go ahead and mark this as resolved.